django/chat/utils.py

```diff
@@ -224,13 +229,21 @@ def sse_string(message: str, format=True, dots=False, remove_stop=False) -> str:
                         "library_str": chat.options.qa_library.name,
                     },
                 )
-                yield sse_string(full_message, format=False, dots=dots_html)
+                yield sse_string(
+                    full_message, format=False, dots=dots_html, remove_stop=remove_stop
+                )
                 await asyncio.sleep(1)
                 first_message = False
-            full_message = response
-            if cache.get(f"stop_response_{message_id}", False):
-                break
-            yield sse_string(full_message, format, dots)
+
+            if remove_stop or not cache.get(f"stop_response_{message_id}", False):
```

Comment:
Even with remove_stop = True, the stop button will appear VERY briefly before the sse_string logic that removes it goes into effect.
Thus, we have to include it as a condition here in order to bypass the below logic for translation/Q+A uploads.